org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalRequestor.java

```diff
@@ -518,7 +495,7 @@ protected boolean isFiltered(CompletionProposal proposal) {
 
 	protected boolean isTypeFiltered(CompletionProposal proposal) {
 		char[] declaringType = getDeclaringType(proposal);
-		return declaringType != null && !this.isImported(declaringType) && TypeFilter.isFiltered(declaringType);
+		return declaringType != null && TypeFilter.isFiltered(declaringType);
```

Comment:
This PR doesn't fix the issue at https://github.com/redhat-developer/vscode-java/issues/2943.

If the proposal is an import completion, we can skip all type filters.

https://github.com/eclipse/eclipse.jdt.ls/blob/f704b7a0f27bc7e693ff7fff5aa678f945c5bf5f/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalReplacementProvider.java#L1060

Reply:
Sounds good. If using this util method, then the `MatcherStringEx` is no needed anymore.